### PR TITLE
runk: Support without pid ns

### PR DESCRIPTION
**Pull request by chethanah**
## Issue
- https://github.com/kata-containers/kata-containers/issues/7168

## Description 
- runk fails to launch OCI container when no PID ns is given in spec (`config.json`)
- This PR fixes runk to launch container when no pid ns is given
```
# jq .linux.namespaces config.json 
[
  {
    "type": "ipc"
  },
  {
    "type": "uts"
  },
  {
    "type": "cgroup"
  },
  {
    "type": "network"
  },
  {
    "type": "mount"
  }
]

# ./runk --log ./runk.log --debug run test
Hello from runk
```

### CODE REVIEW

**Comment by ManaSugi** [comment]
@chethanah Thanks!

Could you fix the title of your commit message by following [the format](https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#general-format)?

Example:
`runk: Support without pid namespace`

**Comment by chethanah** [comment]
> Example: `runk: Support without pid namespace`
I've updated the commit log and pushed the latest changes

**Review by amshinde — APPROVED** [review]

**Review by GabyCT — APPROVED** [review]

File: src/agent/rustjail/src/container.rs [hunk 1]
```diff
@@ -1273,7 +1273,15 @@ fn get_pid_namespace(logger: &Logger, linux: &Linux) -> Result<Option<RawFd>> {
         }
     }
 
-    Err(anyhow!("cannot find the pid ns"))
+    cfg_if::cfg_if! {
+        // if pid ns is not found in spec continue
```
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1277]: What does `continue` mean? Is it a typo?

**Comment by chethanah** [comment]
Hello All,

- Updated the patch to support no pid ns
- pid ns not specified in spec file 
```
# jq .linux.namespaces config.json
[
  {
    "type": "ipc"
  },
  {
    "type": "uts"
  },
  {
    "type": "cgroup"
  },
  {
    "type": "network"
  },
  {
    "type": "mount"
  }
]
```

## Previously...
- In previous changes, runk used to create new ns even if no pid ns was specified
  - but runk should inherit host pid ns, if no ns is specified in sepc file
```
# ./runk --debug --log ./runk.log run test
# ./runk list
ID    PID      STATUS   BUNDLE                               CREATED                            OWNER
test  1283303  running  /home/chethan/projects/bundles/runk  2023-07-10 02:27:27.164110414 UTC  root
# pstree -psS 1283303
systemd(1)───sleep(1283303,cgroup,ipc,mnt,net,pid,uts)
# ls -las /proc/1283303/ns/pid /proc/$$/ns/pid
0 lrwxrwxrwx 1 root root 0 Jul 10 07:57 /proc/1283187/ns/pid -> 'pid:[4026531836]'
0 lrwxrwxrwx 1 root root 0 Jul 10 07:57 /proc/1283303/ns/pid -> 'pid:[4026532405]'
```
- notice new pid ns created.

## Current changes ...
- With latest changes, launch OCI container using runk (based on @ManaSugi changes)
```
# ./runk --debug --log ./runk.log run test
# ./runk list
ID    PID      STATUS   BUNDLE                               CREATED                            OWNER
test  1283885  running  /home/chethan/projects/bundles/runk  2023-07-10 02:31:45.564142770 UTC  root
# pstree -psS 1283885
systemd(1)───sleep(1283885,cgroup,ipc,mnt,net,uts)

# ls -las  /proc/1283885/ns/pid /proc/$$/ns/pid
0 lrwxrwxrwx 1 root root 0 Jul 10 07:57 /proc/1283187/ns/pid -> 'pid:[4026531836]'
0 lrwxrwxrwx 1 root root 0 Jul 10 08:01 /proc/1283885/ns/pid -> 'pid:[4026531836]'

# PID   USER     TIME  COMMAND
    1 root      3:20 {systemd} /sbin/init splash
    2 root      0:00 [kthreadd]
    3 root      0:00 [rcu_gp]
    4 root      0:00 [rcu_par_gp]
    5 root      0:00 [slub_flushwq]
    6 root      0:00 [netns]
    8 root      0:00 [kworker/0:0H-ev]
   11 root      0:00 [mm_percpu_wq]
   12 root      0:00 [rcu_tasks_kthre]
   13 root      0:00 [rcu_tasks_rude_]
...<SKIP>...
```
- runk can list host pid 
- runk has joined the host pid ns when no pid is mentioned in spec file

**Comment by ManaSugi** [comment]
@liubin @GabyCT @amshinde 
Could you review the update again? thanks!

**Comment by ManaSugi** [comment]
@chethanah Could you add `Fixes: #` to your commit message?

**Comment by chethanah** [comment]
> @chethanah Could you add `Fixes: #` to your commit message?

Updated commit message

**Pull request merged**